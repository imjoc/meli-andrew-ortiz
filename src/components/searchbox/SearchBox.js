import React, { Component } from "react";

if (process.env.WEBPACK) require("./searchbox.scss");

class SearchBox extends Component {

	submit () {
		const value = document.querySelector("#search-box").value;
		this.props.submit(value);
	}

	render () {

		return (
			<form className="box-search" onSubmit={(event) => {
				event.preventDefault();
				this.submit();
			}}>
				<input
					type="text"
					name="search"
					maxLength={100}
					placeholder="Nunca dejes de buscar"
					id="search-box"
					defaultValue={this.props.value}
				/>

				<button type="submit">
					<i className="icon-search"><span>Buscar</span></i>
				</button>
			</form>
		);
	}
}

export default SearchBox;