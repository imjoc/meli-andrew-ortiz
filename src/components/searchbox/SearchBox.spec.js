import React from "react";
import {shallow, configure} from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import renderer from "react-test-renderer";
import SearchBox from "./index";
configure({ adapter: new Adapter() });

it("renders searchbox correctly", () => {
    const tree = renderer.create(
        <SearchBox submit="https://api.mercadolibre.com" value="Search" />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});


test("Check render searchbox", function() {
    const component = shallow(
        <SearchBox submit="https://api.mercadolibre.com" value="Search" />
    );
    
    expect(component.text()).toEqual("Buscar");
});
