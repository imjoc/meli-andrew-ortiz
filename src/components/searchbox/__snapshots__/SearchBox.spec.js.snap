// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders searchbox correctly 1`] = `
<form
  className="box-search"
  onSubmit={[Function]}
>
  <input
    defaultValue="Search"
    id="search-box"
    maxLength={100}
    name="search"
    placeholder="Nunca dejes de buscar"
    type="text"
  />
  <button
    type="submit"
  >
    <i
      className="icon-search"
    >
      <span>
        Buscar
      </span>
    </i>
  </button>
</form>
`;
